drivers/rtc/rtc_emul.c

```diff
@@ -277,42 +265,41 @@ static int rtc_emul_set_time(const struct device *dev, const struct rtc_time *ti
 		return -EINVAL;
 	}
 
-	k_mutex_lock(&data->lock, K_FOREVER);
-
-	data->datetime = (*timeptr);
-	data->datetime.tm_isdst = -1;
-	data->datetime.tm_nsec = 0;
+	K_SPINLOCK_GUARDED_SECTION(&data->lock)
```

Comment:
again, great pattern but it should go into a separate tree-wide PR. For now, please use the cumbersome manual approach with `k_spin_lock()`, `k_spin_unlock()`.

Reply:
Ignored for now as not being part of this PR - see discussion below.